test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusDevModeTest.java

```diff
@@ -228,7 +228,7 @@ public void afterEach(ExtensionContext extensionContext) throws Exception {
                 FileUtil.deleteDirectory(deploymentDir);
             }
         }
-        rootLogger.removeHandler(inMemoryLogHandler);
+        inMemoryLogHandler.clearRecords();
```

Comment:
Not exactly sure why we used to remove the in-memory log handler in `afterEach`, because that means capturing the log could only possibly have worked in the first test to be executed within one `QuarkusDevModeTest` instance. All the other remaining tests, should there be any, would see the captured log from the previous test, which seems obviously wrong.

Clearing the remembered content of the in-memory log handler, but leaving attached to the root logger, seems like what should have been done since the beginning, but I'm not sure if there are any consequences I can't see.

Reply:
/cc @stuartwdouglas @geoand

Reply:
Yeah, this seems wrong. FWIW, `QuarkusProdModeTest` already does what @Ladicek did here